### PR TITLE
Added Google Scholar

### PR DESCRIPTION
**Pull request by k4u5h1k**
#### Resolves
Added google scholar module

#### Changes
- Added scholar module that uses lxml

#### Checklist
- [X] I have read the Contribution & Best practices Guideline.
- [X] My branch is up-to-date with the Upstream `master` branch.
- [ ] The acceptance, integration, unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: maryam/core/util/search/scholar.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
```
Comment (saeeddhqan) [inline maryam/core/util/search/scholar.py]: remove re

File: maryam/core/util/search/scholar.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		self.framework.verbose('Searching the scholar.google.com domain...')
+		for start in range(0, (self.max//10)+1):
+			url = f"https://scholar.google.com/scholar"
+			payload = {'as_vis': '1', 'q': self.q, 'start': start}
+			try:
+				req = self.framework.request(
+						url=url,
+						params=payload)
+			except:
+				self.framework.error('[SCHOLAR] ConnectionError', 'util/scholar', 'run_crawl')
```
Comment (saeeddhqan) [inline maryam/core/util/search/scholar.py]: util/search/scholar

File: maryam/core/util/search/scholar.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+				self.framework.error('Google Scholar is missed!', 'util/scholar', 'run_crawl')
```
Comment (saeeddhqan) [inline maryam/core/util/search/scholar.py]: util/search/scholar

File: maryam/core/util/search/scholar.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		return self._articles
+
+	@property
+	def links_with_data(self):
+		findlink = lambda x: x.xpath('h3/a')[0].attrib['href']
+		findauthors = lambda x: x.find_class('gs_a')[0].text_content()
+		findtitle = lambda x: x.xpath('h3/a')[0].text_content()
+
+		for article in self._articles:
+			self._links_with_data.append({
```
Comment (saeeddhqan) [inline maryam/core/util/search/scholar.py:64]: without description?

Comment (k4u5h1k) [reply maryam/core/util/search/scholar.py:64]: Thought it would look cleaner. Have added

File: maryam/core/util/search/scholar.py (hunk 3 above)
Comment (k4u5h1k) [reply maryam/core/util/search/scholar.py]: Corrected. But its still util/whatever is all modules?

**Pull request merged**